tests/GitRelease.py

```diff
@@ -38,148 +38,246 @@
 import os
 import zipfile
 
+from github import GithubException
+
 from . import Framework
 
 
-class Release(Framework.TestCase):
-    def setUp(self):
-        super().setUp()
-        # Do not get self.release here as it casues bad data to be saved in --record mode
-        self.content_path = "content.txt"
-        self.artifact_path = "archive.zip"
+class FileLikeStub:
+    def __init__(self):
+        self.dat = b"I wanted to come up with some clever phrase or something here to test with but my mind is blank."
+        self.file_length = len(self.dat)
+        self.index = 0
 
-        with open(self.content_path, "w") as zip_content:
-            zip_content.write("Pedro for president.")
+    def read(self, size=-1):
+        if size < 0 or size is None:
+            start = self.index
+            self.index = self.file_length
+            return self.dat[start:]
+        else:
+            start = self.index
+            end = start + size
+            self.index = end
+            return self.dat[start:end]
 
-        artifact = zipfile.ZipFile(self.artifact_path, "w")
-        artifact.write(self.content_path)
-        artifact.close()
 
-    def tearDown(self):
-        if os.path.exists(self.content_path):
-            os.remove(self.content_path)
-        if os.path.exists(self.artifact_path):
-            os.remove(self.artifact_path)
-        super().tearDown()
+repo_name = "RepoTest"
+user = "rickrickston123"
+release_id = 28524234
+author_id = 64711998
+tag = "v1.0"
+create_date = datetime.datetime(2020, 7, 12, 7, 34, 42)
+publish_date = datetime.datetime(2020, 7, 14, 0, 58, 20)
+
 
+class ReleaseRead(Framework.TestCase):
+    """
+    Tests that only involve reading, not modification. All data comes from an existing release manually crafted.
```

Comment:
We don't tend to use docstrings for test classes.

Reply:
I'd also prefer this to be ``class GitRelease`` and do everything in that one class.